addons/product/migrations/9.0.1.2/pre-migration.py

```diff
@@ -125,3 +125,11 @@ def migrate(env, version):
     cr.execute(
         "update product_pricelist set active=False where type='purchase'"
     )
+
+    # if weight_net != 0.0 in product_template, override weight
+    # if not, do nothing
+    openupgrade.logged_query(cr, """
+            UPDATE product_template
+            SET weight = weight_net
+            WHERE weight_net != '0.0'
```

Comment:
mmm, adding
<code> and weight_net is not Null </code>
give me the same amount of record, and I have many records null

Reply:
Sorry, I did not realize that NULL != 0.0 evaluates to `NULL` instead of `true`. That means that null values are ignored anyway and you can ignore my comment.